content/articles/python-metaclasses-for-metaprogramming/index.md

```diff
@@ -0,0 +1,233 @@
+---
+layout: engineering-education
+status: publish
+published: true
+url: /python-metaclasses-for-metaprogramming/
+title: Understanding Metaclasses in Python
+description: This tutorial discusses how to use python's metaclasses for metaprogramming. This process involves writing code that manipulates other code. 
+author: dickson-gitau
+date: 2021-12-23T00:00:00-14:10
+topics: [Languages]
+excerpt_separator: <!--more-->
+images:
+
+  - url: /engineering-education/python-metaclasses-for-metaprogramming/hero.jpg
+    alt: Python Metaclasses for Metaprogramming Hero Image
+---
+Metaprogramming is the practice of writing code that manipulates other code. It refers to the ability of a program to regulate itself.
+<!--more-->
+In Python,  metaprogramming can be implemented using decorators and metaclasses.
+
+Object-oriented programming(OOP) is essential in Python. This concept relies on metaprogramming. 

```

Comment:
Form a complete paragraph from these sentences. It is unprofessional to have one line paragraphs